pkg/recipes/terraform/config/config.go

```diff
@@ -124,6 +153,72 @@ func (cfg *TerraformConfig) AddProviders(ctx context.Context, requiredProviders
 		cfg.Provider = providerConfigs
 	}
 
+	// Update module configuration with aliased provider names, if they exist.
+	logger.Info("Updating module providers with aliases")
+	if err := cfg.updateModuleWithProviderAliases(requiredProviders); err != nil {
+		return err
+	}
+
+	// Set the required providers for the Terraform configuration.
+	logger.Info("Update Terraform configuration with required providers")
+	if cfg.Terraform == nil {
+		cfg.Terraform = &TerraformDefinition{}
+	}
+	cfg.Terraform.RequiredProviders = requiredProviders
+
+	return nil
+}
+
+// updateModuleWithProviderAliases updates the module provider configuration in the Terraform config
+// by adding aliases to the provider configurations.
+func (cfg *TerraformConfig) updateModuleWithProviderAliases(requiredProviders map[string]*RequiredProviderInfo) error {
+	if cfg == nil {
+		return fmt.Errorf("terraform configuration is not initialized")
+	}
+	providerConfigs := cfg.Provider
```

Comment:
Do we need to initialize this additional variable? `providerConfigs` seems to be used only once on L181